modules/edw_event/modules/edw_event_agenda/src/Form/MeetingDocumentForm.php

```diff
@@ -98,9 +98,18 @@ public function formRedirect(array &$form, FormStateInterface $form_state) {
     $phase = $this->currentRequest->get('field_document_phase');
     $phase = $this->getDocumentPhase($phase, $form_state);
     $agendaId = $this->currentRequest->get('field_agenda');
-    $options = [
-      'fragment' => "$agendaId",
-    ];
+    $formDocType = $form_state->getValue('field_document_types') ?? [];
+    $docTypeId = $this->currentRequest->get('field_document_types') ?? (!empty($formDocType)
+      ? reset($formDocType)['target_id'] : NULL);
+    $options = [];
+    if (!empty($agendaId)) {
+      $options['fragment'] = "$agendaId";
+    }
+    elseif (!empty($docTypeId)) {
```

Comment:
Maybe you can add this ``if`` first so you can avoid [ElseExpression](https://phpmd.org/rules/cleancode.html#elseexpression)